packages/create-cli/src/utils/directory.ts

```diff
@@ -31,5 +41,14 @@ export function usePackageName (packageName: string) {
   )
 }
 export function hasGitIgnore () {
-  return fs.existsSync('./.gitignore')
+  return fs.existsSync(path.join(process.cwd(), '.gitignore'))
+}
+
+export function isValidUrl (string: string) {
+  try {
+    const url = new URL(string)
+    return url.protocol === 'http:' || url.protocol === 'https:'
+  } catch (_) {
```

Comment:
Hardcore nit (feel free to ignore): our node support allows [to drop `(_)`](https://2ality.com/2017/08/optional-catch-binding.html)